cmd/mattermost/commands/plugin.go

```diff
@@ -0,0 +1,181 @@
+	Use:     "delete",
+	Short:   "Delete a plugin",
+	Long:    "Delete a plugin",
+	Example: `  plugin delete hovercardexample`,
+	RunE:    pluginDeleteCmdF,
+}
+
+var PluginEnableCmd = &cobra.Command{
+	Use:     "enable",
+	Short:   "Enable a plugin",
+	Long:    "Enable a plugin or multiple plugins",
+	Example: `  plugin enable hovercardexample.tar.gz pluginexample.tar.gz`,
+	RunE:    pluginEnableCmdF,
+}
+
+var PluginDisableCmd = &cobra.Command{
+	Use:     "disable",
+	Short:   "Disable a plugin",
+	Long:    "Disable a plugin",
+	Example: `  plugin disable hovercardexample`,
+	RunE:    pluginDisableCmdF,
+}
+
+var PluginListCmd = &cobra.Command{
+	Use:     "list",
+	Short:   "List plugins",
+	Long:    "List all active and inactive plugins",
+	Example: `  plugin list`,
+	RunE:    pluginListCmdF,
+}
+
+func init() {
+	PluginCmd.AddCommand(
+		PluginCreateCmd,
+		PluginDeleteCmd,
+		PluginEnableCmd,
+		PluginDisableCmd,
+		PluginListCmd,
+	)
+	RootCmd.AddCommand(PluginCmd)
+}
+
+func pluginCreateCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	for i, plugin := range args {
+		fileReader, err := os.Open(plugin)
+		if err != nil {
+			return err
+		}
+
+		if _, err := a.InstallPlugin(fileReader); err != nil {
+			return errors.New("Unable to create plugin: " + args[i] + ". Error: " + err.Error())
+		}
+		fileReader.Close()
+	}
+
+	CommandPrettyPrintln("Created plugin(s)")
+
+	return nil
+}
+
+func pluginDeleteCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	if err := a.RemovePlugin(args[0]); err != nil {
+		return errors.New("Unable to delete plugin: " + args[0] + ". Error: " + err.Error())
+	}
+
+	CommandPrettyPrintln("Deleted plugin")
+
+	return nil
+}
+
+func pluginEnableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	for i, plugin := range args {
+		if err := a.EnablePlugin(plugin); err != nil {
+			return errors.New("Unable to enable plugin: " + args[i] + ". Error: " + err.Error())
+		}
+	}
+
+	CommandPrettyPrintln("Enabled plugin(s)")
+
+	return nil
+}
+
+func pluginDisableCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Expected at least one argument. See help text for details.")
+	}
+
+	if err := a.DisablePlugin(args[0]); err != nil {
```

Comment:
I would expect "consistent" behavior here, the same behavior than in `pluginEnableCmdF`. I mean, receive a list of plugins to disable, and threat each element of the list.

Reply:
@jespino from the original ticket, I understood the behavior to be:
* install/enable **multiple** plugins at a time
* delete/disable **one** plugin at a time.

Can you confirm this @esethna?

Reply:
Fair point. I think is inconsistent, but, in fact is what is define in the ticket. @esethna What do you think about it, which is the correct approach here?